current/runtime/storage.md

```diff
@@ -4,22 +4,128 @@ lang: en
+can be used to store lists of items, but runtime developers should take care with respect to the size of these lists.
+Large lists incur storage costs just like large `structs`. Furthermore, iterating over a large list in your runtime may
+result in exceeding the block production time.
+
+#### Methods
+
+Refer to the Storage Value documentation for
+[a comprehensive list of the methods that Storage Values expose](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageValue.html#required-methods).
+Some of the most important methods are summarized here:
+
+##### `get()`
+
+Load the value from storage.
+
+##### `put(val)`
+
+Store the provided value.
+
+##### `mutate(fn)`
+
+Mutate the value with the provided function.
+
+##### `take()`
+
+Remove the value from storage.
+
+### Storage Maps
+
+Storage Maps are implemented as hash maps, which is a pattern that should be familiar to most developers. In order to
+give blockchain engineers increased control over the way in which these data structures are stored, Substrate allows
+developers to select the hashing algorithm that is used to generate map keys. Map data structures are ideal for managing
+sets of items whose elements will be accessed randomly, as opposed to iterating over them sequentially in their entirety.
+
+#### Methods
+
+[Storage Maps expose an API](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageMap.html#required-methods)
+that is similar to that of Storage Values. The selected methods referenced below all use a single key; use two keys for
+[Storage Double Maps](https://substrate.dev/rustdocs/master/frame_support/storage/trait.StorageDoubleMap.html#required-methods):
+
+##### `get(key)`
+
+Load the value associated with the provided key from storage.
+
+##### `store(key, val)`
+
+Store the provided value by associating it with the given key.
+
+##### `mutate(key, fn)`
+
+Use the provided function to mutate the value associated with the given key.
+
+##### `take(key)`
+
+Remove the value associated with the given key from storage.
+
+### Iterable Storage Maps
+
+The Substrate storage API provides iterable map implementations. Because maps are often used to track unbounded sets of
+data (account balances, for example) it is especially likely to exceed block production time by iterating over maps in
+their entirety within the runtime. Furthermore, because maps are comprised of more layers of indirection than native
+lists, they are significantly more costly than lists to iterate over with respect to time. Depending on
+[the hashing algorithm](#Transparent-Hashing-Algorithms) that you select to generate a map's keys, you may be able to
+iterate across its keys as well as its values.
+
+#### Methods
+
+[Iterable Storage Maps expose the following methods](https://substrate.dev/rustdocs/master/frame_support/storage/trait.IterableStorageMap.html#required-methods)
+in addition to the other map methods:
```

Comment:
their method are slightly different for map and doublemap

Reply:
Tried to clarify this in the docs :+1: